### ISSUE TITLE
False-positive for `N811` for full-caps, but non-constant imports

### ISSUE THREAD

**Issue by L4rryFisherman** [astral-sh/ruff#11862]
The line:
```
from uuid import UUID as UUIDType
```
flags as [`N811`](https://docs.astral.sh/ruff/rules/constant-imported-as-non-constant/), whereas UUID is a class and not a constant.

Command used:
```
> ruff check
file.py:1:18: N811 Constant `UUID` imported as non-constant `UUIDType`
```

Ruff version: `0.4.8`

**Comment by zawsq** [comment]
Ruff doesn't check type, it only check if it's caps. 
Just mark it with `noqa`

**Comment by mmarras** [comment]
same false positive here:

`from django.db.models import Q as Query`

imho N811 should not be applied to single character names, as naming conventions for single character names are ambigous anyways and in this particular case cannot distinguish figure out that N801 applies here. Q is a class and flags the above with N811.

**Comment by snowdrop4** [comment]
Addressing the false positive for `from uuid import UUID as UUIDType` is tricky. Technically it might be a violation of PEP8, although a lot people would probably consider it okay. You could potentially whitelist common all-caps words like UUID, but obviously that's fragile, and I don't think anyone wants to go there.

I do, however, have a [PR](https://github.com/astral-sh/ruff/pull/14584) for the following false-positives:

`from foo.bar import A as Apple`
`from foob.bar import Apple as A`
